### PR TITLE
DB-11904 Cache committed transactions (follow up fixes for AZ-2011)

### PR DESCRIPTION
**Pull request by dgomezferro**

### CODE REVIEW

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by msirek — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1  All code was previously reviewed, so can be merged without a second review.

**Pull request merged**
